### ISSUE TITLE
copy & paste is again not working

### ISSUE THREAD

**Issue by fansari** [zellij-org/zellij#1998]
I tried to copy&paste from my Linux Gnome shell but it is not working anymore.

As far as I remember it has worked with the middle button of the mouse.

I have set this:

mouse_mode = true
copy_command "wl-copy -p"

When I copy a text I see the string "Text piped to external command".

When I run "wl-paste --primary" in the console I get the copied text.

So maybe this is some Gnome / Wayland issue - I am unsure.

mouse_mode = false

I can copy & paste but as far as I remember there was some reason not to use it this way (this "extra spaces at the end of the lines" issue). I have tested and this issue is still there.

https://github.com/zellij-org/zellij/issues/1330

Tested with v0.33.0 on Fedora 37 Sliverblue.

**Comment by tlinford** [comment]
Hey, the problem here is that since mouse_mode is on, the middle click is being interpreted by zellij and doesn't cause a paste.

We need to figure out a better way to do this, but for now I think the only workaround in this scenario is to use shift+middle click.

**Comment by fansari** [comment]
It is strange - today just the middle click works as before.

Next time it is not working I will try shift+middle click.

**Comment by tlinford** [comment]
Could the difference be pasting into zellij, vs pasting into another program (this should work fine)?

**Comment by fansari** [comment]
In both cases I tested to paste into Visual Studio Code. I tried again now and it is still working.

**Comment by naosense** [comment]
@tlinford i see ToggleMouseMode action in code base, but there is no bind key in config.kdl, may i ask how to enable it?

**Comment by non-descriptive** [comment]
Zellij 0.37.0 still not working, but yet it says it copied.
If I select with shift and shift+MMB it works, but I it's not zellij doing, but terminal's, since default Gnome Terminal, Xfce Terminal and Xterm expose the same behavior.

**Comment by tlinford** [comment]
> Zellij 0.37.0 still not working, but yet it says it copied. If I select with shift and shift+MMB it works, but I it's not zellij doing, but terminal's, since default Gnome Terminal, Xfce Terminal and Xterm expose the same behavior.

@non-descriptive Please have a look here: https://zellij.dev/documentation/faq.html#copy--paste-isnt-working-how-can-i-fix-thi

**Comment by non-descriptive** [comment]
@tlinford partially it solves the issue. Installed `xclip`, provided external command and provided `copy_clipboard`. It allows to insert via `Ctrl`+`Shift`+`V`, but not `Shift`+`Insert` regardless of provided clipboard buffer. `copy_on_select` set to `true`. 

Also noticed that if I detach from session after changing config it will verify it on attach but will not apply changes. Is it intentional or should I make another issue?

**Comment by tlinford** [comment]
Yeah, at the moment, zellij either copies to clipboard or to primary buffer, not both. I'm thinking I should get back to that soon :)

I'm not sure about the second part, but it may be worth checking if there is already an issue open for that.
